pkg/catalog/catalog.go

```diff
@@ -77,25 +93,16 @@ func (c *ContextCatalog) Upsert(plugin *cli.PluginInfo) error {
 	// When inserting the "global" or "k8s" target we should remove any similar plugin using
 	// the "unknown" target and vice versa.
 	if plugin.Target == configtypes.TargetGlobal || plugin.Target == configtypes.TargetK8s {
-		c.deleteOldTargetEntries(PluginNameTarget(plugin.Name, configtypes.TargetUnknown))
+		delete(c.plugins, PluginNameTarget(plugin.Name, configtypes.TargetUnknown))
 	} else if plugin.Target == configtypes.TargetUnknown {
 		// This could be a global plugin or a k8s plugin (but not both), so let's
 		// delete either pre-existing entry from the catalog
-		c.deleteOldTargetEntries(PluginNameTarget(plugin.Name, configtypes.TargetGlobal))
-		c.deleteOldTargetEntries(PluginNameTarget(plugin.Name, configtypes.TargetK8s))
+		delete(c.plugins, PluginNameTarget(plugin.Name, configtypes.TargetUnknown))
```

Comment:
This line and the one below are the same.
I believe these two lines should not use `configtypes.TargetUnknown` but should use the `configtypes.TargetGlobal` and `configtypes.TargetK8s` like before.

I had to think hard to remember which case this addressed.  Maybe you want to add a better comment here.  Something like:

```
// An older plugin binary may not specify a target (through its 'info' command).
// Therefore the plugin could be a global plugin or a k8s plugin (but not both).
// We need to delete either pre-existing entries from the catalog to avoid having
// a double entry.
```

Reply:
Thanks for catching this. Fixed.